packages/core/admin/admin/src/content-manager/history/components/VersionContent.tsx

```diff
@@ -1,221 +1,90 @@
+import { InputRenderer } from '../../pages/EditView/components/InputRenderer';
 import { useHistoryContext } from '../pages/History';
 
-// These types will be added in future PRs, they need special handling
+/* -------------------------------------------------------------------------------------------------
+ * CustomInputRenderer
+ * -----------------------------------------------------------------------------------------------*/
+
+// The renderers for these types will be added in future PRs, they need special handling
 const UNSUPPORTED_TYPES = ['media', 'relation'];
 
+const CustomInputRenderer = (props: React.ComponentPropsWithoutRef<typeof InputRenderer>) => {
+  if (UNSUPPORTED_TYPES.includes(props.type)) {
+    return <Typography>TODO: support {props.type}</Typography>;
+  }
+
+  return <InputRenderer {...props} />;
+};
+
+/* -------------------------------------------------------------------------------------------------
+ * VersionContent
+ * -----------------------------------------------------------------------------------------------*/
+
 const VersionContent = () => {
+  const [{ query }] = useQueryParams();
   const { version, layout } = useHistoryContext('VersionContent', (state) => ({
     version: state.selectedVersion,
     layout: state.layout,
   }));
-  const [{ query }] = useQueryParams();
-  const dispatch = useTypedDispatch();
-
-  const { runHookWaterfall } = useStrapiApp();
-  const mutatedLayout = runHookWaterfall(HOOKS.MUTATE_EDIT_VIEW_LAYOUT, { layout, query });
-
-  React.useEffect(() => {
-    if (mutatedLayout.layout) {
-      dispatch(setLayout(mutatedLayout.layout, query));
-    }
-  }, [dispatch, mutatedLayout, query]);
-
-  const { permissions } = useSyncRbac(query, layout.contentType.uid, 'editView');
-
-  const { readActionAllowedFields } = getFieldsActionMatchingPermissions(
-    permissions ?? [],
-    mutatedLayout.layout!.contentType.uid
-  );
-
-  const getCustomFields = React.useCallback(() => {
-    if (!mutatedLayout.layout) {
-      return [];
-    }
-
-    const customFields: string[] = [];
-    Object.values(mutatedLayout.layout.contentType.attributes).forEach((value) => {
-      if ('customField' in value) {
-        customFields.push(value.customField as string);
-      }
-    });
-
-    Object.values(mutatedLayout.layout.components).forEach((component) => {
-      Object.values(component.attributes).forEach((value) => {
-        if ('customField' in value) {
-          customFields.push(value.customField as string);
-        }
-      });
-    });
-
-    return customFields;
-  }, [mutatedLayout.layout]);
-
-  const { isLazyLoading, lazyComponentStore } = useLazyComponents(getCustomFields());
-  const customInputs = useCustomInputs(lazyComponentStore);
-
-  // TODO: better loading
-  if (isLazyLoading) {
-    return null;
-  }
-
-  const layoutPanes = splitLayoutIntoPanes(mutatedLayout.layout);
+  const { permissions = [] } = useSyncRbac(version.contentType, query, 'VersionContent');
```

Comment:
If we don't provide permissions, NotAllowInput gets rendered everywhere. Are you saying there's a better way to get the permissions?

Reply:
i mean't specifically `useSyncRbac` just because it's some weird thing that lets i18n run permission middleware, but that won't run because i18n won't inject them unless the container is `editView` or `listview`. You could just use `useRBAC`?

Reply:
I'm a bit confused because useRBAC takes `permissions` (the thing we need) as a prop.

It seems to me like useRbac does the logic to check whether the user has the permissions we pass it. In our case that's not what we want, the checking logic is done by the InputRenderer component.

What we want is the array of Permission objects, and I'm not sure where else I would get it

Reply:
I think you could get it straight from the redux store, but to be honest that's super confusing too and is different for each place 🤪 lets leave your solution as is, I think it's fine ⭐